rero_ils/modules/items/serializers/collector.py

```diff
@@ -0,0 +1,433 @@
+        """Append library data.
+
+        :param libraries_map: libraries mapping.
+        :param csv_data: data dictionary.
+        """
+        csv_data['item_library_name'] = libraries_map[
+            csv_data.pop('item_library_pid')]
+
+    @classmethod
+    def append_location_data(cls, csv_data, locations_map):
+        """Append location data.
+
+        :param locations_map: locations mapping.
+        :param csv_data: data dictionary.
+        """
+        csv_data['item_location_name'] = locations_map[
+            csv_data.pop('item_location_pid')]
+
+    @classmethod
+    def append_document_data(cls, csv_data, documents):
+        """Append document data.
+
+        :param csv_data: data dictionary.
+        :param documents: documents data.
+        """
+        try:
+            # update csv data with document
+            csv_data.update(documents.get(csv_data.get('document_pid')))
+        except Exception as err:
+            current_app.logger.error(
+                'ERROR in csv serializer: '
+                '{message} on document: {pid}'.format(
+                    message=err,
+                    pid=hit['document']['pid'])
+            )
+
+    @classmethod
+    def append_item_local_fields(cls, csv_data):
+        """Append item local fields data.
+
+        :param csv_data: data dictionary.
+        """
+        org_pid = csv_data.get('item_org_pid')
+        if local_fields := LocalField.get_local_fields_by_resource(
+                'item', csv_data.get('item_pid'), organisation_pid=org_pid):
+            for field, num in itertools.product(local_fields, range(1, 11)):
+                field_name = f'item_local_field_{num}'
+                if field_data := field.get('fields', {}).get(f'field_{num}'):
+                    csv_data[field_name] = ' '.join(filter(None, field_data))
+
+    @classmethod
+    def append_document_local_fields(cls, csv_data):
+        """Append document local fields data.
+
+        :param csv_data: data dictionary.
+        """
+        org_pid = csv_data.pop('item_org_pid')
+        if local_fields := LocalField.get_local_fields_by_resource(
+                'doc', csv_data.get('document_pid'), organisation_pid=org_pid):
+            for field, num in itertools.product(local_fields, range(1, 11)):
+                field_name = f'document_local_field_{num}'
+                if field_data := field.get('fields', {}).get(f'field_{num}'):
+                    csv_data[field_name] = ' '.join(filter(None, field_data))
+
+    @classmethod
+    def get_loans_by_item_pids(cls, item_pids=None):
+        """Get loans for the given item pid list.
+
+        :param item_pids: item pids.
+        :return list of checkouts, renewals, last_transaction_dates,
+                last_checkouts.
+        """
+        checkouts_query = RecordsSearch(index=LoanOperationLog.index_name)\
+            .filter('term', loan__trigger=ItemCirculationAction.CHECKOUT)\
```

Comment:
Mmmm...It seems that `loan.trigger` isn't mapped into ES. It should be no ?
If it's not mapped into ES, it should be analyze as "text" with a raw (keyword) children attribute (@jma : is it correct ?)
So either add this field into ES_mapping (... and create an alembic script to reindex all alive loans), either use `loan__trigger__raw`. (I vote for the first solution !)